docs/tutorials/mkldnn/mkldnn_quantization.md

```diff
@@ -0,0 +1,259 @@
+
+```python
+# quantization configs
+# set exclude layers
+excluded_names = []
+# set calib mode.
+calib_mode = 'naive'
+# set calib_layer
+calib_layer = None
+# set quantized_dtype
+quantized_dtype = 'auto'
+logger.info('Quantizing FP32 model resnet18-V1')
+cqsym, cqarg_params, aux_params, collector = quantize_graph(sym=sym, arg_params=arg_params, aux_params=aux_params,
+                                                          excluded_sym_names=excluded_names,
+                                                          calib_mode=calib_mode, calib_layer=calib_layer,
+                                                          quantized_dtype=quantized_dtype, logger=logger)
+
+# download imagenet validation dataset
+mx.test_utils.download('http://data.mxnet.io/data/val_256_q90.rec', 'dataset.rec')
+# set rgb info for data
+mean_std = {'mean_r': 123.68, 'mean_g': 116.779, 'mean_b': 103.939, 'std_r': 58.393, 'std_g': 57.12, 'std_b': 57.375}
+# set batch size
+batch_size = 16
+# create DataIter
+data = mx.io.ImageRecordIter(path_imgrec='dataset.rec', batch_size=batch_size, data_shape=batch_shape[1:], rand_crop=False, rand_mirror=False, **mean_std)
+# create module
+mod = mx.mod.Module(symbol=sym, label_names=None, context=mx.cpu())
+mod.bind(for_training=False, data_shapes=data.provide_data, label_shapes=None)
+mod.set_params(arg_params, aux_params)
+
+# calibration configs
+# set num_calib_batches
+num_calib_batches = 5
+max_num_examples = num_calib_batches * batch_size
+# monitor FP32 Inference
+mod._exec_group.execs[0].set_monitor_callback(collector.collect, monitor_all=True)
+num_batches = 0
+num_examples = 0
+for batch in data:
+    mod.forward(data_batch=batch, is_train=False)
+    num_batches += 1
+    num_examples += batch_size
+    if num_examples >= max_num_examples:
+        break
+if logger is not None:
+    logger.info("Collected statistics from %d batches with batch_size=%d"
+                % (num_batches, batch_size))
+```
+
+After that, layer information will be filled into the `collector` returned by `quantize_graph` api. Then, you need to write the layer information into int8 model by calling `calib_graph` api.
+
+
+```python
+# write scaling factor into quantized symbol
+cqsym, cqarg_params, aux_params = calib_graph(qsym=cqsym, arg_params=arg_params, aux_params=aux_params,
+                                            collector=collector, calib_mode=calib_mode,
+                                            quantized_dtype=quantized_dtype, logger=logger)
+# (optional) visualize quantized model
+mx.viz.plot_network(cqsym)
+```
+
+Below is a quantized residual block with naive calibration. We can see `min_calib_range` and `max_calib_range` are wrote into `_contrib_requantize` operators.
+
+![naive calibrated model](naive_calib.png)
+
+When you get a quantized model with calibration, keep sure to call fusion api again since this can fuse some `requantize` or `dequantize` operators for further performance improvement.
```

Comment:
my experience has been that without another fusion pass, the accuracy is really bad. Do you confirm?

Reply:
Thanks for your suggestions. I'll take a try.